Simplify handling providers

invidian opened the pull request:
As a prerequisite for simplified version of #137.

rata commented on aws/flatcar-linux/kubernetes/require.tf:
```diff
@@ -2,24 +2,12 @@
-provider "tls" {
-  version = "~> 2.0"
+  required_providers {
+    aws      = ">= 2.31.0"
```
Doesn't this allow version 3.0.0? I don't think we want that, as that would mean the aws provider probably broke compatibility. Why not maintain the symbols used before?

invidian replied:
Oops, bad copy-paste. will fix.

invidian replied:
Fixed.

rata commented on aws/flatcar-linux/kubernetes/require.tf:
```diff
@@ -2,24 +2,12 @@
+    local    = "~> 1.2"
+    null     = "~> 2.1"
+    template = "~> 2.1"
+    tls      = "~> 2.0"
```
no ct provider needs to be added here? (I see you added it on bare-metal)

invidian replied:
It wasn't there, but yeah, it should be added. Adding in a separate commit.

rata commented on packet/flatcar-linux/kubernetes/require.tf:
```diff
@@ -2,32 +2,14 @@
-provider "aws" {
-  version = "2.31.0"
+  required_providers {
+    aws      = "2.31.0"
```
I the think previous (deleted) missing `~>` was probably a bug introduced recently (tf 0.12 migration?)

invidian replied:
Could be, but I'd say it's out of scope for this PR to fix that?

invidian replied:
Created #143 for it.

rata replied:
:ok_hand:

mauriciovasquezbernal replied:
> I the think previous (deleted) missing `~>` was probably a bug introduced recently (tf 0.12 migration?)

It wasn't me :stuck_out_tongue_closed_eyes:

rata commented on packet/flatcar-linux/kubernetes/workers/require.tf:
```diff
@@ -2,24 +2,14 @@
-provider "packet" {
-  version = "~> 2.7.3"
+  required_providers {
+    aws      = "2.31.0"
```
idem

invidian replied:
Created #143 for it.

invidian commented:
Updated:
- I split changes per platform, so they are easier for review.
- Checked all platforms for missing constraints and added them where needed.
- Make sure docs are updated as well.

invidian commented:
Will rebase and merge after #146.
